application/common/middlewares.py

```diff
@@ -16,12 +17,8 @@ def unhandled_exception_logging(handler, event, context):
 
 
 @lambda_handler_decorator(trace_execution=True)
-def set_correlation_id_if_none_set(handler, event, context: LambdaContext):
-
-    correlation_id = logger.get_correlation_id()
-
-    if correlation_id is None:
-        logger.set_correlation_id(context.aws_request_id)
-
+def set_correlation_id(handler, event: SQSEvent, context: LambdaContext):
```

Comment:
The original aim of this function was to set a correlation id of the requestId in the case where a correlationId was not provided, you could still achieve this with the revised by doing. 

```python
correlation_id = next(event.records).message_attributes["correlation-id"]["stringValue"]
if correlation_id is None:
    logger.set_correlation_id(context.aws_request_id)
else: 
 logger.set_correlation_id(correlation_id)
````

Reply:
Correlation Id must be passed into the API-Gateway otherwise the process doesn't work. This isn't a python issue but part of the API Gateway to SQS transformation

Reply:
we cannot force NHSUK to send us a correlation id, but if they do we should use that. the point of the fallback is that if they dont then we still have a correlation id to tie events between us and DoS

Reply:
which it must if the tests still pass so will resolve

Reply:
This is now in the API Gateway which will try to pass in the correlation id but if correlation id is null or "" then uses the API Gateway request id